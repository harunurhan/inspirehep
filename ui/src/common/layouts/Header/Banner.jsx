import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Alert, Button } from 'antd';
import PropTypes from 'prop-types';

import {
  SUBMISSIONS,
} from '../../routes';
import { setBannerVisibility } from '../../../actions/ui';


class Banner extends Component {
  constructor(props) {
    super(props);
    this.onBannerAlertClose = this.onBannerAlertClose.bind(this);
  }

  onBannerAlertClose() {
    const { dispatch } = this.props;
    dispatch(setBannerVisibility(false));
  }

  render() {
    const { shouldDisplayBanner, isSubmissionsPage } = this.props;
    return shouldDisplayBanner && !isSubmissionsPage && (
      <div className="banner">
        <Alert
          type="info"
          banner
          closable
          afterClose={this.onBannerAlertClose}
          showIcon={false}
          message={
            <span>
              <strong>
                INSPIRE beta provides a preview of new features currently under development. Try it out and let us know what you think!
              </strong>
              <Button
                className="ml3"
                type="primary"
                target="_blank"
                href="https://goo.gl/forms/aTYSRzd7vTUhxzL43"
              >
                Take the survey
              </Button>
            </span>
          }
        />
      </div>
    );
  }
}

Banner.propTypes = {
  isSubmissionsPage: PropTypes.bool.isRequired,
  shouldDisplayBanner: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const stateToProps = state => ({
  isSubmissionsPage: String(state.router.location.pathname).startsWith(SUBMISSIONS),
  shouldDisplayBanner: state.ui.get('bannerVisibility'),
});

const dispatchToProps = dispatch => ({ dispatch })

export default connect(stateToProps, dispatchToProps)(Banner);
