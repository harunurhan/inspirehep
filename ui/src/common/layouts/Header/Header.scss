.__Header__ {
  .sticky {
    z-index: $zindex-header;
    position: fixed;
    width: 100%;

    // Non sticky/fixed header for mobile
    @media (max-width: $screen-xs-max) {
      position: initial;
      z-index: initial;
    }

    .header {
      padding: $header-padding;

      @media (max-width: $screen-xs-max) {
        line-height: 56px;
        height: initial; // unset height for mobile
      }
    }

    .intervention-banner {
      text-align: center;
    }
  }

  .non-sticky {
    background-color: $layout-header-background;

    // remove the margin top for mobile
    @media (max-width: $screen-xs-max) {
      margin-top: 0 !important;
    }
  }
}