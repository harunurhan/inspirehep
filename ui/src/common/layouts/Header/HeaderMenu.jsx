import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Menu } from 'antd';

import {
  SUBMISSIONS_AUTHOR,
  USER_LOGIN,
  SUBMISSIONS_JOB,
  SUBMISSIONS_LITERATURE,
  SUBMISSIONS_CONFERENCE,
} from '../../routes';
import ExternalLink from '../../components/ExternalLink';
import LinkLikeButton from '../../components/LinkLikeButton';

import './HeaderMenu.scss';
import { PAPER_SEARCH_URL } from '../../constants';

class HeaderMenu extends Component {
  render() {
    const { loggedIn, onLogoutClick } = this.props;
    return (
      <Menu
        className="__HeaderMenu__"
        theme="dark"
        mode="horizontal"
        selectable={false}
      >
        <Menu.Item key="search-tips">
          <ExternalLink href={PAPER_SEARCH_URL}>Search Tips</ExternalLink>
        </Menu.Item>
        <Menu.SubMenu
          key="submit"
          title="Submit"
          popupClassName="header-submenu"
        >
          <Menu.Item key="submit.author">
            <Link to={SUBMISSIONS_AUTHOR}>Author</Link>
          </Menu.Item>
          <Menu.Item key="submit.job">
            <Link to={SUBMISSIONS_JOB}>Job</Link>
          </Menu.Item>
          <Menu.Item key="submit.literature">
            <Link to={SUBMISSIONS_LITERATURE}>Literature</Link>
          </Menu.Item>
          <Menu.Item key="submit.conference">
            <Link to={SUBMISSIONS_CONFERENCE}>Conference</Link>
          </Menu.Item>
        </Menu.SubMenu>
        <Menu.Item key="login-logout">
          {loggedIn ? (
            // TODO: create LoginLinkOrLogoutButtonContainer
            <LinkLikeButton onClick={onLogoutClick} dataTestId="logout">
              Logout
            </LinkLikeButton>
          ) : (
            <Link to={USER_LOGIN}>Login</Link>
          )}
        </Menu.Item>
      </Menu>
    );
  }
}

HeaderMenu.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  onLogoutClick: PropTypes.func.isRequired,
};

export default HeaderMenu;
