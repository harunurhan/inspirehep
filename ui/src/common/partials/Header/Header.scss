.__Header__ {
  z-index: 2;
  position: fixed;
  width: 100%;

  .menuItems {
    line-height: 64px;
  }
}
