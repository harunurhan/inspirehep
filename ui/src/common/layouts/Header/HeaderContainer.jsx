import { connect } from 'react-redux';

import { SUBMISSIONS, HOME } from '../../routes';
import Header from './Header';

const stateToProps = state => ({
  isHomePage: state.router.location.pathname === HOME,
  isSubmissionsPage: String(state.router.location.pathname).startsWith(
    SUBMISSIONS
  ),
});

export default connect(stateToProps)(Header);
