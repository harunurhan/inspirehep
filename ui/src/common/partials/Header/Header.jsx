import React from 'react';
import { Link } from 'react-router-dom';
import { Layout, Menu, Row, Col } from 'antd';

import SearchBoxContainer from './../../containers/SearchBoxContainer';
import './Header.scss';

const Header = () => (
  <Layout className="__Header__">
    <Layout.Header>
      <Row type="flex" align="middle" gutter={16}>
        <Col span={4}>
          <Link to="/literature">INSPIRE</Link>
        </Col>
        <Col span={16}>
          <SearchBoxContainer />
        </Col>
        <Col span={4}>
          <Menu className="menuItems" theme="dark" mode="horizontal">
            <Menu.Item>
              <Link to="/holdingpen/dashboard">Holdingpen</Link>
            </Menu.Item>
          </Menu>
        </Col>
      </Row>
    </Layout.Header>
  </Layout>
);

export default Header;
