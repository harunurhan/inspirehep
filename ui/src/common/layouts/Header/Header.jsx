import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Layout, Row, Col } from 'antd';
import PropTypes from 'prop-types';

import SearchBoxContainer from '../../containers/SearchBoxContainer';
import './Header.scss';
import Logo from '../../components/Logo';
import {
  SUBMISSIONS,
  HOME,
} from '../../routes';
import Banner from './Banner';
import HeaderMenu from './HeaderMenu';


class Header extends Component {
  render() {
    const { isHomePage, isSubmissionsPage } = this.props;
    return (
      <div className="__Header__">
        <Banner />
        <Layout.Header className="header">
          <Row type="flex" align="middle" gutter={{ xs: 8, md: 16}}>
            <Col xs={0} lg={4} xl={5}>
              <Logo />
            </Col>
            <Col xs={18} lg={12} xl={13} xxl={14}>
              {!isHomePage && !isSubmissionsPage && <SearchBoxContainer />}
            </Col>
            <Col xs={6} lg={8} xl={6} xxl={5}>
              <HeaderMenu />
            </Col>
          </Row>
        </Layout.Header>
      </div>
    );
  }
}

Header.propTypes = {
  isHomePage: PropTypes.bool.isRequired,
  isSubmissionsPage: PropTypes.bool.isRequired,
};

const stateToProps = state => ({
  isHomePage: state.router.location.pathname === HOME,
  isSubmissionsPage: String(state.router.location.pathname).startsWith(SUBMISSIONS),
});

export default connect(stateToProps)(Header);
