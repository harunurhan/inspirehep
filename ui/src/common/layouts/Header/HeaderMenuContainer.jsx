import { connect } from 'react-redux';

import { userLogout } from '../../../actions/user';
import HeaderMenu from './HeaderMenu';

const stateToProps = state => ({
  loggedIn: state.user.get('loggedIn'),
});

const dispatchToProps = dispatch => ({
  onLogoutClick() {
    dispatch(userLogout());
  },
});

export default connect(stateToProps, dispatchToProps)(HeaderMenu);
