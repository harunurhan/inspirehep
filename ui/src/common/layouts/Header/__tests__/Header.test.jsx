import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('Header', () => {
  it('renders with search box if it is not on home', () => {
    const wrapper = shallow(
      <Header isSubmissionsPage={false} isHomePage={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without search box if it is on homepage `/`', () => {
    const wrapper = shallow(<Header isSubmissionsPage={false} isHomePage />);
    expect(wrapper).toMatchSnapshot();
  });
});
