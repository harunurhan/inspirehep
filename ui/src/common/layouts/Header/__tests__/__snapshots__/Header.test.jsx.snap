// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders with search box if it is not on home 1`] = `
<div
  className="__Header__"
>
  <Connect(Banner) />
  <Adapter
    className="header"
  >
    <Row
      align="middle"
      gutter={
        Object {
          "md": 16,
          "xs": 8,
        }
      }
      type="flex"
    >
      <Col
        lg={4}
        xl={5}
        xs={0}
      >
        <Logo />
      </Col>
      <Col
        lg={12}
        xl={13}
        xs={18}
        xxl={14}
      >
        <Connect(SearchBox) />
      </Col>
      <Col
        lg={8}
        xl={6}
        xs={6}
        xxl={5}
      >
        <Connect(HeaderMenu) />
      </Col>
    </Row>
  </Adapter>
</div>
`;

exports[`Header renders without search box if it is on homepage \`/\` 1`] = `
<div
  className="__Header__"
>
  <Connect(Banner) />
  <Adapter
    className="header"
  >
    <Row
      align="middle"
      gutter={
        Object {
          "md": 16,
          "xs": 8,
        }
      }
      type="flex"
    >
      <Col
        lg={4}
        xl={5}
        xs={0}
      >
        <Logo />
      </Col>
      <Col
        lg={12}
        xl={13}
        xs={18}
        xxl={14}
      />
      <Col
        lg={8}
        xl={6}
        xs={6}
        xxl={5}
      >
        <Connect(HeaderMenu) />
      </Col>
    </Row>
  </Adapter>
</div>
`;
